python/sglang/srt/layers/moe/topk.py

```diff
@@ -207,15 +207,34 @@ def select_experts(
                     topk_group=topk_group,
                 )
         else:
-            topk_weights, topk_ids = biased_grouped_topk(
-                hidden_states=hidden_states,
-                gating_output=router_logits,
-                correction_bias=correction_bias,
-                topk=top_k,
-                renormalize=renormalize,
-                num_expert_group=num_expert_group,
-                topk_group=topk_group,
-            )
+            device = hidden_states.device
+            if device == torch.device("cpu") and cpu_has_amx_support():
+                M = hidden_states.size(0)
+                topk_weights = torch.empty(
+                    M, top_k, dtype=torch.float32, device=device
+                )
+                topk_ids = torch.empty(M, top_k, dtype=torch.int32, device=device)
```

Comment:
we may need to allocate output tensors in C++ instead of python, this is a little slow but too much.